test/integration/daemonset/daemonset_test.go

```diff
@@ -393,3 +411,327 @@ func TestInsufficientCapacityNodeDaemonDoesNotLaunchPod(t *testing.T) {
 		close(stopCh)
 	}
 }
+
+func getPods(t *testing.T, podClient corev1typed.PodInterface, labelMap map[string]string) *v1.PodList {
+	podSelector := labels.Set(labelMap).AsSelector()
+	options := metav1.ListOptions{LabelSelector: podSelector.String()}
+	pods, err := podClient.List(options)
+	if err != nil {
+		t.Fatalf("Failed obtaining a list of pods that match the pod labels %v: %v", labelMap, err)
+	}
+	return pods
+}
+
+func checkDaemonSetPodsTemplateGeneration(podClient corev1typed.PodInterface, podInformer cache.SharedIndexInformer, templateGeneration string, t *testing.T) {
+	if err := wait.Poll(pollInterval, pollTimeout, func() (bool, error) {
+		pods := podInformer.GetIndexer().List()
+		for _, object := range pods {
+			pod := object.(*v1.Pod)
+			// We don't care about inactive pods
+			if !controller.IsPodActive(pod) {
+				continue
+			}
+			podTemplateGeneration := pod.Labels[v1beta1.DaemonSetTemplateGenerationKey]
+			if podTemplateGeneration != templateGeneration {
+				return false, fmt.Errorf("expected pod %s/%s template generation %q, but got %q", pod.Namespace, pod.Name, templateGeneration, podTemplateGeneration)
+			}
+		}
+
+		return true, nil
+	}); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func checkDaemonSetPodsOrphaned(podClient corev1typed.PodInterface, t *testing.T) {
+	if err := wait.Poll(pollInterval, pollTimeout, func() (bool, error) {
+		pods := getPods(t, podClient, testLabels())
+		for _, pod := range pods.Items {
+			// Manually delete the pod's OwnerReferences
+			pod = *updatePod(t, podClient, pod.Name, func(pod *v1.Pod) {
```

Comment:
I had to manually delete the pods' `OwnerReferences` because `DeletePropagationOrphan`:
1. does not orphan the pods in integration test environment (unexpected)
2. does not delete the DaemonSet from key-value store (expected)